pkg/p2p/kadcast/network.go

```diff
@@ -61,5 +63,61 @@ func sendUDPPacket(netw string, addr net.UDPAddr, payload []byte) {
 	if err != nil {
 		log.WithError(err).Warn("Error while writting to the filedescriptor.")
 		return
-	} 
-}
\ No newline at end of file
+	}
+}
+
+// StartTCPListener listens infinitely for TCP packet arrivals and
+// executes it's processing inside a gorutine by sending
+// the packets to the circularQueue.
+func StartTCPListener(netw string, queue *ring.Buffer, MyPeerInfo Peer) {
+	lAddr := getLocalTCPAddress()
+	// Set listening port.
+	lAddr.Port = int(MyPeerInfo.port)
+PacketConnCreation:
+	// listen to incoming TCP packets
+	listener, err := net.ListenTCP(netw, &lAddr)
+	if err != nil {
+		log.Panic(err)
+	}
+	// Set initial deadline.
+	listener.SetDeadline(time.Now().Add(time.Minute))
+
+	// Instanciate the buffer
+	buffer := make([]byte, 5000000)
+	for {
+		// Read TCP packet.
+		pc, err := listener.AcceptTCP()
+		uAddr := pc.RemoteAddr()
+		byteNum, err := io.ReadFull(pc, buffer)
```

Comment:
Do we need to read so many bytes per each wire message (5000000)? If less than 5000000 bytes are received an error will be returned.

Reply:
This was set according to the max block size @toghrulmaharramov suggested

Reply:
Do you mean that by design each message on the wire should be always the max block size? E.g If we send a shorter message like `topic.Tx` with  size of 1000 bytes the rest to 5000000 must be padding?

Reply:
Current Node P2P reader works with length prefixed messages where the wire message length variable.

Reply:
At the time I wrote this, If you sent less bytes than the size of the buffer, no error was returned at all. I tested the code hardly and It was not producing any error.

Reply:
How did you test it?

Regarding TCP as it's a stream protocol, you need to treat the data on wire as a stream. Option one is to use length-prefix frame, option two is to use frame delimiter. That's why  `io.ReadFull` is in use here to ensure we read `exactly len(buf) bytes from r into buf.`

Reagrding io.ReadFull()
```go
// ReadFull reads exactly len(buf) bytes from r into buf.
// It returns the number of bytes copied and an error if fewer bytes were read.
func ReadFull(r Reader, buf []byte) (n int, err error) {
```

Reply:
7 instances on diferent machines syncronizing themseleves and broadcasting CHUNKS messages.

Reply:
It does not work if less bytes are sent. In my attempt to be helpful, I gave the explanation abovementioned. Feel free to retest or ignore all.

Reply:
> Current Node P2P reader works with length prefixed messages where the wire message length is variable up to max block size.

Probably implementing something like this would be the best option.
My knowledge of Go is very limited or None hahahah